Hopefully better approach for testing logger (#4478)
Particular/docs.particular.net: nservicebus/testing/index_content_testing_[6,).partial.md

The following code show how to verify that logging is performed by the message handler.

snippet: LoggerTestingSetup

> The setup fixture above sets the testing logging factory once per assembly because the factory is statically cached for the live time of the application domain. Subsequent test executions then clear the logged statements before every test run as shown below.

Reviewer: ```suggestion
The setup fixture above sets the testing logging factory once per assembly because the factory is statically cached during the lifetime of the application domain. Subsequent test executions then clear the logged statements before every test run as shown below.
```
